bluepysnap/_plotting.py

```diff
@@ -204,7 +207,7 @@ def spikes_isi(filtered_report, use_frequency=False, binsize=None, ax=None):  #
     if binsize is not None and binsize <= 0:
         raise BluepySnapError(f"Invalid binsize = {binsize}. Should be > 0.")
 
-    gb = filtered_report.report.groupby(["ids", "population"])
+    gb = filtered_report.report.groupby(["ids", "population"], observed=False)
```

Comment:
`observed=False` is to keep the current behavior and to silence the pandas warning that it will default to `observed=True` in the future.

However, I'm not convinced we should keep observed=False.

Reply:
From what I can see, it doesn't make any difference when `filtered_report` is a spike report, because `filtered_report.report` returns a dataframe with columns `ids` and `population`, where only `population` is categorical, and it's explicitly converted to category [here](https://github.com/BlueBrain/snap/blob/6315a6b8299c4d5396c84fbfb84c146b4c5b7b86/bluepysnap/spike_report.py#L175)

So it's not possible to have categories not observed in the actual data.
I think we can set `observed=True`, that will be the default in a future version, and maybe add a short comment to explain the reason.

Reply:
From what I can see, it doesn't make difference when `filtered_report` is a spike report, because `filtered_report.report` returns a dataframe with columns `ids` and `population`, where only `population` is categorical, and it's explicitly converted to category [here](https://github.com/BlueBrain/snap/blob/6315a6b8299c4d5396c84fbfb84c146b4c5b7b86/bluepysnap/spike_report.py#L175)

So it's not possible to have categories not observed in the actual data.
I think we can set `observed=True`, that will be the default in a future version, and maybe add a short comment to explain the reason.